### PR TITLE
Fix update qty of cart when qty is set to 0 on front-office

### LINKED ISSUE #25470: Chrome - FO - Quantity increment +1 when to we change quantity to 0 in cart with press enter key
#### Describe the bug

In the shopping cart, when we change the quantity to 0 with **press enter Key** using Chrome. We have the quantity changes to 1 then to 2.
And I remark that i put a quantity she's multiplied by 2. 
Is possible bug to js ? 
https://drive.google.com/file/d/1dNzXFy-HEoM3BxJGJ2-snIjId08ZYzbu/view?usp=sharing
#### Expected behavior

We have checked on the official demonstration website prestashop and we have found the same issue. 

#### Additional information

* PrestaShop version: last version

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | When you update the product qty to 0 with the keyboard, product is having a strange behavior
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25470.
| How to test?      | Add a product to the cart, update the qty to 10 manually by typing it and press enter, it should be fine, update, do the same thing but with 0 => the product should be removed
| Possible impacts? | Cart qty update

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25883)

### CODE REVIEW

**Comment by matks** [comment]
Re running CI

**Review by matthieu-rolland — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by marwachelly** [comment]
Hello,
The following steps are not working as expected :

1. add  product  with minimum quantity for sale equal to 15 (different from1) 
2. Add a product to the cart, update the quantity to a value < 15  manually by typing it and press enter => the quantity is multiplied => Not ok 

https://www.awesomescreenshot.com/video/5272159?key=afe7c44462b8e4e2af896ba6475a2f37

Thanks,

**Comment by NeOMakinG** [comment]
@marwachelly should be fine now :)

File: themes/classic/_dev/js/cart.js [hunk 1]
```diff
@@ -194,8 +194,8 @@ $(document).ready(() => {
         let dataset;
 
         if ($target && $target.dataset) {
-          // eslint-disable-next-line
-          dataset = $target.dataset;
+        // eslint-disable-next-line
+        dataset = $target.dataset;
```
Comment (PierreRambaud) [inline themes/classic/_dev/js/cart.js]: Nope :scream:

Comment (NeOMakinG) [reply themes/classic/_dev/js/cart.js]: Meh, my bad

**Review by PierreRambaud — DISMISSED** [review]

File: themes/classic/_dev/js/cart.js [hunk 2]
```diff
@@ -247,19 +247,27 @@ $(document).ready(() => {
...
+    if (targetValue === '0') {
+      $target.closest('.product-line-actions').find('[data-link-action="delete-from-cart"]').click();
+    } else {
+      $target.attr('value', targetValue);
+      sendUpdateQuantityInCartRequest(updateQuantityInCartUrl, getRequestData(qty), $target);
+    }
   }
 
   $body.on('focusout keyup', productLineInCartSelector, (event) => {
     if (event.type === 'keyup') {
```
Comment (Progi1984) [inline themes/classic/_dev/js/cart.js:259]:
```suggestion
    isUpdateOperation = false;
    if (event.type === 'keyup') {
```

**Review by Progi1984 — COMMENTED** [review]
@NeOMakinG Thank you for your contribution. I have brought a small comment.

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by marwachelly** [comment]
Hello @NeOMakinG ,
I go to the BO and I fixed the minimum quantity of a product  to 15 then from the FO I Added the product to the cart, I updated the qty to a number < 15 manually by typing it and I pressed enter, the quantity changed to anorther number different from 15.
https://www.awesomescreenshot.com/video/5761106?key=fee6ef68a3ad3f0272d43b6316fb9360
Could you please check.
Thanks,

**Comment by NeOMakinG** [comment]
@marwachelly did you build assets and avoid any caches? Seems like a cache issue, it should be fixed :/

**Comment by NeOMakinG** [comment]
I would like another QA test if possible, I can't find why it's not working :/

**Comment by marwachelly** [comment]
@PrestaShop/qa-functional I retested the PR again but I still have the same issue. Could you please check?

**Comment by khouloudbelguith** [comment]
Hi @NeOMakinG, @marwachelly,

I manage to reproduce the issue of minimal quantity.
Steps to reproduce the behavior:
1. Create a product with minimal quantity = 5
image
2. Go to FO > Product details page > add the product to the cart
3. On the Shopping cart page
4. Change quantity to 2 and click enter with the keyboard
5. See error => the quantity changed automatically

I used PHP version 7.2
I used Chrome v93.0.4577.63 (Build officiel) (64 bits)

I attached a screen record
https://watch.screencastify.com/v/ZhY75ap3BEi5NUy3AIbT

Thanks!

**Comment by NeOMakinG** [comment]
Gonna take a look then... so strange :(

**Comment by NeOMakinG** [comment]
@khouloudbelguith @marwachelly 

I took the branch, built assets, then tried to test on a product with min quantity of 5 =>

https://www.awesomescreenshot.com/video/5907614?key=e65b31ca2893f19ad572e18b1efe4b38

I can't reproduce the bug 🤣

**Comment by NeOMakinG** [comment]
Did you try it with your console opened with cache disabled inside network tab?

**Comment by marwachelly** [comment]
Hello,
The issue is fixed using Firefox browser but the same issue is reproducible with Chrome 92+

image

Could you please check?
Thanks,

**Comment by NeOMakinG** [comment]
> Hello, The issue is fixed using Firefox browser but the same issue is reproducible with Chrome 92+
> 
> image
> 
> Could you please check? Thanks,

image

I use this version of chrome and can't reproduce, but I doubt this is linked to chrome 92 honestly

**Comment by NeOMakinG** [comment]
@marwachelly I was testing on the branch without the review feedback form @Progi1984 which was causing a buggy behavior.

It looks like it's fixed now, I hope so, and my apologies 😄

**Comment by marwachelly** [comment]
> @marwachelly I was testing on the branch without the review feedback form @Progi1984 which was causing a buggy behavior.
> 
> It looks like it's fixed now, I hope so, and my apologies smile

Hope that the issue will be fixed, thanks a lot.

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello,

PR is tested and the issue is fixed. The following case are tested:
- minimum quantity of product is 0 
- minimum quantity is different from 0
- product simple/ customized product/ pack of product.

Thanks,

**Comment by NeOMakinG** [comment]
Thanks @marwachelly

**Pull request merged**